actioncable/lib/action_cable/server/base.rb

```diff
@@ -92,16 +119,28 @@ def worker_pool
         @worker_pool || @mutex.synchronize { @worker_pool ||= ActionCable::Server::Worker.new(max_size: config.worker_pool_size) }
       end
 
+      # Executor is used by various actions within Action Cable (e.g., pub/sub operations) to run code asynchronously.
+      def executor
+        @executor || @mutex.synchronize { @executor ||= ThreadedExecutor.new(max_size: config.executor_pool_size) }
+      end
+
       # Adapter used for all streams/broadcasting.
       def pubsub
-        @pubsub || @mutex.synchronize { @pubsub ||= config.pubsub_adapter.new(self) }
+        @pubsub || (executor && @mutex.synchronize { @pubsub ||= config.pubsub_adapter.new(self) })
       end
 
       # All of the identifiers applied to the connection class associated with this
       # server.
       def connection_identifiers
         config.connection_class.call.identifiers
       end
+
+      # Tags are declared in the server but computed in the connection. This allows us per-connection tailored tags.
+      # You can pass request object either directly or via block to lazily evaluate it.
```

Comment:
This doesn't seem lazy at the moment?

Reply:
The block is only called if any of the `config.log_tags` is a Proc. Otherwise, we don't need to initialize a request object. So, the "lazy" refers to the difference between:

```ruby
new_tagged_logger(request)

# and
new_tagged_logger { request }
```